src/Microsoft.OData.Core/UriParser/Parsers/ODataPathParser.cs

```diff
@@ -1432,5 +1437,150 @@ private void CheckDollarEachSegmentRestrictions(int index)
                 }
             }
         }
+
+        private void CheckTypeCastSegmentRestriction(ODataPathSegment previous, IEdmType targetEdmType)

```

Comment:
>CheckTypeCastSegmentRestriction [](start = 21, length = 31)

We should also validate, if the previous segment is an operation, that the return type of the operation can be cast to the specified type.

Reply:
Separate offline discussion on how to support annotating ReturnType.

---
In reply to: [236788996](https://github.com/OData/odata.net/pull/1295#discussion_r236788996) [](ancestors = 236788996)